source/background.js

```diff
@@ -79,51 +81,59 @@ async function update() {
 	}
 }
 
-const handleBrowserActionClick = async () => {
+async function handleBrowserActionClick() {
 	await openTab(await getTabUrl());
-};
+}
 
 function handleInstalled(details) {
 	if (details.reason === 'install') {
 		browser.runtime.openOptionsPage();
 	}
 }
 
-function handleConnectionStatus(event) {
-	if (event.type === 'online') {
+function handleConnectionStatus() {
+	if (navigator.onLine) {
 		update();
-	} else if (event.type === 'offline') {
+	} else {
 		handleOfflineStatus();
 	}
 }
 
-window.addEventListener('online', handleConnectionStatus);
-window.addEventListener('offline', handleConnectionStatus);
-
-browser.alarms.create({when: Date.now() + 2000});
-browser.alarms.onAlarm.addListener(update);
-browser.runtime.onMessage.addListener(message => {
+function onMessage(message) {
 	if (message === 'update') {
 		update();
 	}
-});
+}
 
 async function addNotificationHandler() {
 	if (await queryPermission('notifications')) {
 		browser.notifications.onClicked.addListener(id => {
 			openNotification(id);
 		});
 
-		browser.notifications.onClosed.addListener(id => {
-			removeNotification(id);
-		});
+		// Possbily broken in browser, event listener getting fired if not displayed to user
```

Comment:
either fix the issue, or remove the commented out code, we have complete file history in git for this.

Reply:
I should've put this PR as a draft. This is intended as a test PR. It's late here so I thought I'd update it in the morning.